src/components/engagement/pnp-extractor.service.ts

```diff
@@ -16,12 +16,47 @@ export class PnpExtractor {
     @Logger('pnp:extractor') private readonly logger: ILogger
   ) {}
 
+  private year = 0;
+  private quarter = 0;
+
+  private getYearAndQuarterFromFileName(fileName: string) {
+    // first check for 2021, 2020, 2019
+    // the reason I did this is b/c some file names had multiple sets of 4 numbers (2020 and 2003 for instance)
```

Comment:
Maybe you could match all years and then take the latest one?

Reply:
Yeah. That would be more solid.

Reply:
This gets a bit tricky though because 4 digits may not be a year. I can also check that the number isn't greater than 2021, but still not bulletproof.

Reply:
What about?
```tsx
const possibleYears = [];

const currentYear = DateTime.local().year;
const years = possibleYears.filter(year => year <= currentYear && year > currentYear - 30);
```

If you could distill all the options we are trying to account for it would be trivial to make a test for it.

Reply:
Yeah. This code is going away before too long though, right?

Reply:
Yeah don't spend too much time on it